puma/matshow.py

```diff
@@ -0,0 +1,198 @@
+
+        self.mat = matrix
+        self.x_ticklabels = x_ticklabels
+        self.x_ticks_rotation = x_ticks_rotation
+        self.y_ticklabels = y_ticklabels
+        self.show_entries = show_entries
+        self.show_percentage = show_percentage
+        self.text_color_threshold = text_color_threshold
+        self.colormap = colormap
+        self.cbar_label = cbar_label
+        self.atlas_offset = atlas_offset
+
+        # Specifying figsize if not specified by user
+        if self.figsize is None:
+            self.figsize = (10, 10.5)
+        self.initialise_figure()
+        self.__plot()
+
+    def __get_luminance(self, rgbaColor):
+        """Calculate the relative luminance of a color according to W3C standards.
+        For the details of the conversion see: https://www.w3.org/WAI/GL/wiki/Relative_luminance .
+
+        Parameters
+        ----------
+        rgbColor : tuple
+            (r,g,b,a) color (returned from `plt.cm` colormap)
+
+        Returns
+        -------
+        float
+            Relative luminance of the color.
+        """
+        # Converting to np.ndarray, ignoring alpha channel
+        rgbaColor = np.array(rgbaColor[:-1])
+        rgbaColor = np.where(
+            rgbaColor <= 0.03928, rgbaColor / 12.92, ((rgbaColor + 0.055) / 1.055) ** 2.4
+        )
+        weights = np.array([0.2126, 0.7152, 0.0722])
+        return np.dot(rgbaColor, weights)
+
+    def __plot(self):
+        """Plot the Matrix."""
+        im = self.axis_top.matshow(self.mat, cmap=self.colormap)
+
+        # If mat entries have to be plotted
+        if self.show_entries:
+            # Mapping mat values in [0,1], as it's done by matplotlib
+            # to associate them to the colors of the colormap
+            normMat = self.mat - np.min(self.mat)
+            normMat = normMat / (np.max(self.mat) - np.min(self.mat))
+
+            # Adding text values in the matrix pixels
+            for i in range(self.mat.shape[0]):
+                for j in range(self.mat.shape[1]):
+                    # Choosing the text color: black if color is light, white if color is dark
+                    # Getting the bkg color from the cmap
+                    color = self.colormap(normMat[i, j])
+                    # Calculating the bkg relative luminance
+                    luminance = self.__get_luminance(color)
+                    # Choosing the appropriate color
+                    color = "white" if luminance <= self.text_color_threshold else "black"
+
+                    # Value of the matrix, eventually converted to percentage
+                    text = (
+                        str(round(self.mat[i, j], 3))
+                        if not self.show_percentage
+                        else str(round(self.mat[i, j] * 100, 2)) + "%"
+                    )
+                    # Plotting text
+                    self.axis_top.text(
+                        x=j, y=i, s=text, va="center", ha="center", size="x-large", c=color
```

Comment:
Can we take font size from the `PlotBase` properties?
In order to make it customizable